src/platforms/android/androidiaphandler.cpp

```diff
@@ -168,45 +169,140 @@ void AndroidIAPHandler::onNoPurchases(JNIEnv* env, jobject thiz) {
   Q_UNUSED(thiz);
   // ToDo - I'm not sure when this scenario would occur
   // and so I'm not sure what the best way to handle it is.
-  logger.debug() << "onNoPurchases event occured";
+  logger.info() << "onNoPurchases event occured";
 }
 
 // static
-void AndroidIAPHandler::subscriptionFailed(JNIEnv* env, jobject thiz) {
+void AndroidIAPHandler::onSubscriptionFailed(JNIEnv* env, jobject thiz) {
   Q_UNUSED(env)
   Q_UNUSED(thiz);
-  IAPHandler::instance()->stopSubscription();
-  IAPHandler::instance()->subscriptionFailed();
-  IAPHandler::instance()->productsRegistered();
+  IAPHandler* iap = IAPHandler::instance();
+  iap->stopSubscription();
+  emit iap->subscriptionFailed();
+}
+
+// static
+void AndroidIAPHandler::dispatchToMainThread(std::function<void()> callback) {
```

Comment:
@strseb I copied this code from the AndroidWebView because I need to make a network request and that needs to happen in the main thread (that i wasn't in after coming back from kt code). Should this just be a generic android util. Or am I misusing this code and I don't need a timer or something else.

Reply:
AndroidUtils is probably the right place for it now :)